src/node/txreconciliation.cpp

```diff
@@ -53,6 +90,17 @@ class TxReconciliationState
      */
     uint64_t m_k0, m_k1;
 
+    /**
+     * Store all wtxids which we would announce to the peer (policy checks passed, etc.)
```

Comment:
Have you considered to enforce policy checks like feefilter at `REQTXRCNCL` sending instead of at reconciliation set fulfilling in INV sending ?

Due to `RECON_REQUEST_INTERVAL`, I think some transactions could become stalled, wasting bandwidth in case of fast-paced mempool congestion change (not to exclude when we see few transactions issuers on the network provoking spikes by themselves). Though if correct, more likely a marginal performance improvement better left for follow-up.

Reply:
This would largely complicate the code I think.... Policy check stuff should be moved out of `net_processing.cpp`.
Yeah, I see your concern (although I tested Erlay in mainnet and it was ok), let's see if other reviewers think it's needed now.

Reply:
To clarify, we're considering the scenario in which our peer's fee filter changes between adding tx to `txs_to_reconcile` and sending the `inv` after reconciliation? Given that `RECON_REQUEST_INTERVAL` is 8sec and `AVG_FEEFILTER_BROADCAST_INTERVAL` is 10min, I don't think this is too much of a concern?

Reply:
I thought this could happen even without explicit fee filter changes, but rather the following:
1. A transaction marginally passed the given fee filter.
2. In 8 seconds when it's time to send, it doesn't pass the filter, but is sent anyway.